tests/lib/rules/func-name-matching.js

```diff
@@ -22,19 +22,43 @@ ruleTester.run("func-name-matching", rule, {
     valid: [
         { code: "var foo;" },
         { code: "var foo = function foo() {};" },
+        { code: "var foo = function foo() {};", options: ["always"] },
+        { code: "var foo = function bar() {};", options: ["never"] },
         { code: "var foo = function() {}"},
         { code: "var foo = () => {}", parserOptions: { ecmaVersion: 6} },
         { code: "foo = function foo() {};" },
+        { code: "foo = function foo() {};", options: ["always"] },
+        { code: "foo = function bar() {};", options: ["never"] },
         { code: "obj.foo = function foo() {};" },
+        { code: "obj.foo = function foo() {};", options: ["always"] },
+        { code: "obj.foo = function bar() {};", options: ["never"] },
         { code: "obj.foo = function() {};" },
+        { code: "obj.foo = function() {};", options: ["always"] },
+        { code: "obj.foo = function() {};", options: ["never"] },
         { code: "obj.bar.foo = function foo() {};" },
+        { code: "obj.bar.foo = function foo() {};", options: ["always"] },
+        { code: "obj.bar.foo = function bar() {};", options: ["never"] },
```

Comment:
Any chance the function in this test case could be renamed, so that there's no chance of confusion that the object name must be the match here? (e.g., `obj.bar.foo = function baz() {};`)